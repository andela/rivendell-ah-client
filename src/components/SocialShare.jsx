import React from 'react';
import { Button } from 'semantic-ui-react';
import propTypes from 'prop-types';

const SocialShare = (props) => {
  const { articleURL } = props;

  const socialMessage = encodeURIComponent(
    'Here is a cool'
  + ' article I just found in author\'s'
  );
  const twitterUrl = 'https://twitter.com/intent/tweet?url='
  + `${articleURL}
&text=${socialMessage}
&hashtags=${encodeURIComponent('AuthorsHaven')}`;

  const facebookUrl = `https://www.facebook.com/dialog/share?
app_id=${process.env.FACEBOOK_ID}
&display=page
&href=${articleURL}&redirect_uri=${articleURL}`;


  const linkedinUrl = `https://www.linkedin.com/shareArticle?mini=true&
url=${articleURL}&summary=${socialMessage}&source=Author'sHaven`;

  return (
    <div id="social-share">
      <div>
        <a href={twitterUrl}>
          <Button
            name="twitter"
            key="twitter-btn"
            icon="twitter"
            content="Share on Twitter"
            color="twitter"
          />
        </a>

        <a href={facebookUrl}>
          <Button
            key="facebook-btn"
            icon="facebook"
            content="Share on Facebook"
            color="facebook"
            name="facebook"
          />
        </a>

        <a href={linkedinUrl}>
          <Button
            key="linkedin-btn"
            icon="linkedin"
            content="Share on Linkedin"
            color="linkedin"
            name="linkedin"
          />
        </a>

      </div>
    </div>

  );
};


SocialShare.propTypes = {
  articleURL: propTypes.string.isRequired,
};

export default SocialShare;
