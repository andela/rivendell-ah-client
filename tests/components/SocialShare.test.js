import { shallow } from 'enzyme';
import chai from 'chai';
import React from 'react';
import { Button } from 'semantic-ui-react';
import SocialShare from '../../src/components/SocialShare';


const getMountedObj = (articleUrl = '') => shallow(
  <SocialShare articleUrl={articleUrl} />
);
describe('Testing the SocialShare component', () => {
  describe('the components rendered', () => {
    it('should render three anchor tags', () => {
      chai.expect(getMountedObj().find('a').length)
        .to.equal(3);
    });
    it('should render three Button(s) tags', () => {
      chai.expect(getMountedObj().find(Button).length)
        .to.equal(3);
    });
    it('should match snapshots', () => {
      expect(getMountedObj()).toMatchSnapshot();
    });
  });

  describe('the props of the anchors ', () => {
    it('there should be an anchor twiiter link', () => {
      chai.expect(getMountedObj()
        .find('a[href^="https://twitter.com/intent/tweet"]').length)
        .to.equal(1);
    });
    it('there should be an anchor with facebook link', () => {
      chai.expect(getMountedObj()
        .find('a[href^="https://www.facebook.com/dialog/share?"]').length)
        .to.equal(1);
    });
    it('there should be an anchor with facebook link', () => {
      chai.expect(getMountedObj()
        .find('a[href^="https://www.linkedin.com/shareArticle?"]').length)
        .to.equal(1);
    });
  });
});
