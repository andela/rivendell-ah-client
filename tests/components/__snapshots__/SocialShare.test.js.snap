// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the SocialShare component the components rendered should match snapshots 1`] = `
<div
  id="social-share"
>
  <div>
    <a
      href="https://twitter.com/intent/tweet?url=undefined
&text=Here%20is%20a%20cool%20article%20I%20just%20found%20in%20author's
&hashtags=AuthorsHaven"
    >
      <Button
        as="button"
        color="twitter"
        content="Share on Twitter"
        icon="twitter"
        key="twitter-btn"
        name="twitter"
        role="button"
      />
    </a>
    <a
      href="https://www.facebook.com/dialog/share?
app_id=undefined
&display=page
&href=undefined&redirect_uri=undefined"
    >
      <Button
        as="button"
        color="facebook"
        content="Share on Facebook"
        icon="facebook"
        key="facebook-btn"
        name="facebook"
        role="button"
      />
    </a>
    <a
      href="https://www.linkedin.com/shareArticle?mini=true&
url=undefined&summary=Here%20is%20a%20cool%20article%20I%20just%20found%20in%20author's&source=Author'sHaven"
    >
      <Button
        as="button"
        color="linkedin"
        content="Share on Linkedin"
        icon="linkedin"
        key="linkedin-btn"
        name="linkedin"
        role="button"
      />
    </a>
  </div>
</div>
`;
